src/linear_algebra/affine_space/basic.lean

```diff
@@ -1226,6 +1259,24 @@ rfl
   (f -ᵥ g : affine_map k P1 V2) p = f p -ᵥ g p :=
 rfl
 
+/-- `prod.fst` as an `affine_map`. -/
+def fst : affine_map k (P1 × P2) P1 :=
+{ to_fun := prod.fst,
+  linear := linear_map.fst k V1 V2,
+  map_vadd' := λ _ _, rfl }
```

Comment:
Can you use
```suggestion
def fst : affine_map k (P1 × P2) P1 := linear_map.fst.to_affine_map
```

Reply:
No, `linear_map.to_affine_map` reinterprets a `linear_map` as an `affine_map` between modules, and I want `to_fun` to be defeq `prod.fst`.

Reply:
Isn't `f.to_affine_map.to_fun` defeq to `f`, and `linear_map.fst.to_fun` defeq to `prod.fst`? Or is that not enough?

Reply:
`linear_map.to_affine_map` reinterprets `f : V1 →ₗ[k] V2` as an `affine_map k V1 V2`, not `affine_map k P1 P2`. While I can add `linear_map.to_affine_map' (f : V1 →ₗ[k] V2) (p1 : P1) (p2 : P2) : affine_map k P1 P2 := { to_fun := λ p, f (p -ᵥ p1) +ᵥ p2, ... }`, it will produce wrong defeq in this case.